nengo_gui/static/components/spa_similarity.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * Line graph showing semantic pointer decoded values over time
+ * @constructor
+ *
+ * @param {dict} args - A set of constructor arguments (see Nengo.Component)
+ * @param {int} args.n_lines - number of decoded values
+ * @param {float} args.min_value - minimum value on y-axis
+ * @param {float} args.max_value - maximum value on y-axis
+ * @param {Nengo.SimControl} args.sim - the simulation controller
+ */
+
+Nengo.SpaSimilarity = function(parent, sim, args) {
+    Nengo.Value.call(this, parent, sim, args);
+
+    this.data_store = new Nengo.VariableDataStore(this.n_lines, this.sim, args.synapse);
+    this.show_pairs = false;
+
+    var self = this;
+
+    this.colors = Nengo.make_colors(6);
+    this.color_func = function(d, i){return self.colors[i%6]};
```

Comment:
This line could use some more whitespace like so
`this.color_func = function(d, i) { return self.colors[i % 6] };